compiler/ic/navigator.nim

```diff
@@ -102,7 +102,13 @@ proc nav(g: ModuleGraph) =
   # translate the track position to a packed position:
   let unpacked = g.config.m.trackPos
   let mid = unpacked.fileIndex
-  let fileId = g.packed[int32 mid].fromDisk.sh.strings.getKeyId(toFullPath(g.config, mid))
+  let sharedPart = g.packed[int32 mid].fromDisk.sh
+
+  # sharedPart being nil is stand-in for incorrect tracking info specified
+  if sharedPart == nil:
+    globalError(g.config, unpacked, "Invalid track info provided")
```

Comment:
I fail to see the connection. Why would ``.sh`` be nil anyway?

Reply:
If the trackPos fileIndex is for a non-existent file then:
1. `g.packed[int32 mid]` with have a .`status` of `undefined`
2. `g.packed[int32 mid].fromDisk` will return an uninitialized `PackedModule`
3. `.sh` will be `nil` as it's a reference

That's quite the garden path, but that's because I'm not quite sure what the invariants are at this particular stage. My current mental model, still developing is that I should be able to assume (but I'm not sure):

* anything that calls `nav` assumes that things have quiesced
* if the modulegraph doesn't know about a module now, it's not going to know about it period as far as we're concerned

So:
* does that imply that a `LoadedModule` with `.status` of `undefined` implies a bad file name?
* would it be better to check `.module` for a `nil` instead?

I'm still getting up to speed on the procs, so I might have just missed it. 😬

Reply:
undefined entries are entries belonging to filenames that are not Nim modules, such as configuration files or include files. `.sh` would be nil for an inlcude file too, which **is** a valid tracking position. However, this currently cannot work XD

We need a solution for include files first, and then if it's neither an include file nor a module we can produce an error.